drivers/i2c/i2c_nrf5.c

```diff
@@ -279,6 +290,8 @@ static int i2c_nrf5_init(struct device *dev)
 	volatile NRF_TWI_Type *twi = config->base;
 	int status;
 
+	k_sem_init(&data->lock, 1, 1);
```

Comment:
use K_SEM_DEFINE() in line 354 (for instance):

static struct i2c_nrf5_data i2c_nrf5_data_0 = {
K_SEM_DEFINE(lock, 1, 1),
};

I think it should work. 

Do the same for port 1

Reply:
What's wrong with k_sem_init() I do the same as @tkln in other drivers

Reply:
It's a tiny optimization to use K_SEM_DEFINE. I can dismiss my review, it's not a blocker after all.

Reply:
It's not wrong in the sense of being incorrect, but the semaphore does get initialized twice, once statically to zero and then properly by k_ksem_init(). Initializing it correctly statically has no runtime overhead and is probably preferrable.

Reply:
@tbursztyka understood. @tkln OK, feel free to update the PR before we merge it then

Reply:
@tbursztyka @carlescufi  Now that I tried to implement it, it seems that K_SEM_DEFINE() cannot be used with struct members. It looks like K_SEM_INITIALIZER() would do what we need, but it has apparently been deprecated.

Reply:
You can use _K_SEM_INITIALIZER(), see commit 65a9d2a94a

Idea is to restrict usage of these macros to kernel space, and you driver is in kernel space so it's fine.